Sources/Swift/Integrations/SessionReplay/SentryReplayOptions.swift

```diff
@@ -216,39 +278,108 @@ public class SentryReplayOptions: NSObject, SentryRedactOptions {
      *  - errorSampleRate Indicates the percentage in which a 30 seconds replay will be send with
      * error events.
      */
-    public init(sessionSampleRate: Float = 0, onErrorSampleRate: Float = 0, maskAllText: Bool = true, maskAllImages: Bool = true, enableViewRendererV2: Bool = false, enableFastViewRendering: Bool = false) {
+    public init(
+        sessionSampleRate: Float = 0,
+        onErrorSampleRate: Float = 0,
+        maskAllText: Bool = true,
+        maskAllImages: Bool = true,
+        enableViewRendererV2: Bool = true,
+        enableFastViewRendering: Bool = false
+    ) {
+        // Note for future maintainers:
+        // - This initializer is publicly available for Swift, but not for Objective-C.
```

Comment:
I think it is common knowledge that Swift initializers with default values are not available to ObjC, we don't need to write this in every spot.

Additionally, if it is indeed intended that we allow consumers to instantiate an instance of SentryReplayOptions from objc, we should code that in the iOS-ObjectiveC sample app to avoid breaking things in the future.

Reply:
Yes, common knowledge if you have background in using Objective-C. But in my case I didn't know why the method was there in the first place, because Sesssion Replay is mostly written and used in Swift (in the SDK), but then I realized the constructor is there because of `SentryOptions.m`.

Reply:
I think it would be good to comment on why it's not available in ObjC.